src/exoticatechnologies/modifications/exotics/impl/GuardianShield.kt

```diff
@@ -0,0 +1,711 @@
+                // Move the drone to center of ship
+                drone?.let {
+                    it.location.set(ship.location.x, ship.location.y)
+                    it.facing = ship.facing
+                }
+
+                shieldController.assesSituation(amount)
+                shieldController.controlShield(amount)
+                shieldController.shieldPushOutEffect(amount)
+                shieldController.transferFlux(amount, workingMode = FluxTransferMode.ONE_WAY, transfer = USE_TRANSFER)
+            }
+        }
+
+        private fun systemOn() {
+            // Enable turning the shield on *only* if neither the drone nor the host ship are overloaded or venting
+            //
+            // Had to be added because the shield was "flashing" / "flickering" (tried turning on before cancelling)
+            // while the host ship was being overloaded
+            drone?.let {
+                if (it.fluxTracker.isOverloadedOrVenting.not() && ship.fluxTracker.isOverloadedOrVenting.not()) {
+                    it.shield.toggleOn()
+                    log("systemOn()\tshield on ? ${it.shield.isOn}\t\tship's flux dissipation: ${ship.mutableStats.fluxDissipation.modifiedValue}", "$LOGTAG:GuardianShieldDrone")
+                } else {
+                    log("systemOn()\taborted due to overload/venting", "$LOGTAG:GuardianShieldDrone")
+                }
+                if (APPLY_FLUX_DEBUFF) {
+                    ship.mutableStats.fluxDissipation.modifyMult(GUARDIAN_SHIELD_FLUX_DISSIPATION_DEBUFF_ID, FLUX_DEBUFF_AMOUNT)
+                }
+
+                if (APPLY_HARDFLUX_DEBUFF) {
+                    ship.mutableStats.hardFluxDissipationFraction.modifyMult(GUARDIAN_SHIELD_HARDFLUX_DISSIPATION_DEBUFF_ID, HARDFLUX_DEBUFF_AMOUNT)
+                }
+            }
+        }
+
+        private fun systemOff() {
+            drone?.let {
+                it.shield.toggleOff()
+                log("systemOff()\tshield off ? ${it.shield.isOff}\t\tship's flux dissipation: ${ship.mutableStats.fluxDissipation.modifiedValue}", "$LOGTAG:GuardianShieldDrone")
+                if (APPLY_FLUX_DEBUFF) {
+                    ship.mutableStats.fluxDissipation.unmodify(GUARDIAN_SHIELD_FLUX_DISSIPATION_DEBUFF_ID)
+                }
+
+                if (APPLY_HARDFLUX_DEBUFF) {
+                    ship.mutableStats.hardFluxDissipationFraction.unmodify(GUARDIAN_SHIELD_HARDFLUX_DISSIPATION_DEBUFF_ID)
+                }
+            }
+        }
+
+        override fun spawnDrone(): ShipAPI {
+            Global.getCombatEngine().getFleetManager(ship.owner).isSuppressDeploymentMessages = true
+
+            val fleetSide = FleetSide.values()[ship.owner]
+            val fighter = CombatUtils.spawnShipOrWingDirectly(
+                    getDroneVariant(),
+                    FleetMemberType.SHIP,
+                    fleetSide,
+                    1f,
+                    ship.location,
+                    ship.facing
+            )
```

Comment:
as suggested, replacing this with
`CombatEngineAPI.createFXDrone` might solve the selectability issue? give that a try.

Reply:
```
    public static ShipAPI spawnDrone(ShipAPI host, String weaponID){
        ShipAPI demDrone;
        ShipHullSpecAPI spec = Global.getSettings().getHullSpec("dem_drone");
        ShipVariantAPI v = Global.getSettings().createEmptyVariant("dem_drone", spec);
        v.addWeapon("WS 000", weaponID);
        WeaponGroupSpec g = new WeaponGroupSpec(WeaponGroupType.LINKED);
        g.addSlot("WS 000");
        v.addWeaponGroup(g);

        demDrone = Global.getCombatEngine().createFXDrone(v);
        demDrone.setLayer(CombatEngineLayers.ABOVE_SHIPS_AND_MISSILES_LAYER);
        demDrone.setOwner(host.getOriginalOwner());
        //demDrone.getMutableStats().getBeamWeaponRangeBonus().modifyFlat("dem", targetingLaserRange);
        demDrone.getMutableStats().getHullDamageTakenMult().modifyMult("dem", 0f); // so it's non-targetable
        demDrone.setDrone(true);
        demDrone.getAIFlags().setFlag(ShipwideAIFlags.AIFlags.DRONE_MOTHERSHIP, 100000f, host);
        //demDrone.getMutableStats().getWeaponTurnRateBonus().modifyMult(id,0);
        //demDrone.getMutableStats().getBeamWeaponTurnRateBonus().modifyFlat("StormbringerBeamEffect", 10000);
        demDrone.setCollisionClass(CollisionClass.NONE);
        demDrone.giveCommand(ShipCommand.SELECT_GROUP, null, 0);
        Global.getCombatEngine().addEntity(demDrone);
        return demDrone;
    }
```

Reply:
ooooh indeed it did. ggwp, so the FXDrone fixed two things!

Reply:
so the FXDrone did indeed fix the clickability thing, except there's a big triangle on the tacmap now. oh well, better than what it used to be still. not idea, but better.

Reply:
resolving